src/app/cluster/details/external-cluster/external-machine-deployment-details/template.html

```diff
@@ -22,8 +22,8 @@
     <mat-card-header>
       <mat-card-title class="machine-deployment-name">
         <i class="machine-deployment-health km-pointer"
-           [ngClass]="getStatusColor()"
-           [matTooltip]="getStatusMessage()"></i>
+           [matTooltip]="machineDeploymentHealthStatus.message"
+           [ngClass]="machineDeploymentHealthStatus.icon"></i>
```

Comment:
```suggestion
           [matTooltip]="machineDeploymentHealthStatus?.message"
           [ngClass]="machineDeploymentHealthStatus?.icon"></i>
```
Because `machineDeploymentHealthStatus` is updated inside `subscribe`  so there's a chance that `machineDeploymentHealthStatus` can be `undefined` when this is rendered.